### DISCUSSION TITLE
graphql/codegen/docs/config-reference/config-field

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#9596] [Docs Discussions]
# graphql/codegen/docs/config-reference/config-field

Learn how to configure GraphQL Code Generator to fit your needs. The `config` field is used to pass configuration to Plugins.

https://the-guild.dev/graphql/codegen/docs/config-reference/config-field

**Comment by konrazem** [comment]
I am looking for full config options explanation like:
- apolloClientVersion: 3
- reactApolloVersion: 3
- withHooks: true
- preResolveTypes: true
- withHOC: false
- withComponent: false
- withMutationFn: false
- skipTypename: false
- avoidOptionals: true
- maybeValue: T | null | undefined

I tihnk it was explained here some time ago 🤔
NOTE: I am using old version 2.16.4

**Reply by salazarm** [reply]
I also am looking for this

**Reply by gkhore** [reply]
Get them here
https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-react-apollo
